mono/metadata/appdomain.c

```diff
@@ -86,8 +86,16 @@
  *
  * Changes which are already detected at runtime, like the addition
  * of icalls, do not require an increment.
+ *
+ * IMPORTANT: when updating the version, use the following format:
+ *    1<major><minor><build><corlibversion> (every group is exactly two digits)
+ *
+ *    e.g. for Mono 4.9.1 and corlib version 32 it'd be: 104090132
+ *
+ *    Whenever Mono's version number is bumped in configure.ac this needs
+ *    to be bumped too. The corlib version can be reset to 00 in that case.
  */
-#define MONO_CORLIB_VERSION 164
+#define MONO_CORLIB_VERSION 104090100
```

Comment:
The `1` prefix is there because in C an integer literal that starts with 0 is defined as octal, which we don't want.

The reason why I think two digits for the corlib version are enough is that we were at `164` before and this mechanism was introduced many years ago. Given that the number will be reset to 00 when we bump the Mono version this should be plenty enough.

And yes, the reason for major being two digits too is that we could end up with two digit major (see the 17.02 discussion).